packages/firestore/src/local/local_store.ts

```diff
@@ -511,155 +513,155 @@ export class LocalStore {
 
     return this.persistence
       .runTransaction('Apply remote event', 'readwrite-primary', txn => {
-        const documentBuffer = this.remoteDocuments.newChangeBuffer({
-          trackRemovals: true // Make sure document removals show up in `getNewDocumentChanges()`
-        });
-
-        // Reset newTargetDataByTargetMap in case this transaction gets re-run.
-        newTargetDataByTargetMap = this.targetDataByTarget;
+        return this.targetCache
```

Comment:
Could you explain a bit of the rationale behind this change? It seems mostly like this is just a rearrangement of what came before, but I'm having a hard time understanding both what changed and why.